src/test/controllerscriptenginelegacy_test.cpp

```diff
@@ -620,3 +636,245 @@ TEST_F(ControllerScriptEngineLegacyTest, connectionExecutesWithCorrectThisObject
+                     engine.setValue('[Test]', 'coTimerId', this.timerId + 10);
+                  }, true);            
+                  engine.setValue('[Test]', 'coTimerId', this.timerId);
+               }
+            }
+            var MyMapping = new MyClass();
+            MyMapping.runTimer();)"));
+    processEvents();
+    EXPECT_DOUBLE_EQ(0.0, co->get());
+    double timerId = coTimerId->get();
+    EXPECT_TRUE(timerId > 0);
+
+    cEngine->thread()->msleep(35);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(timerId + 10, coTimerId->get());
+    EXPECT_DOUBLE_EQ(7.0, co->get());
+    EXPECT_TRUE(evaluateAndAssert("engine.setValue('[Test]', 'co', MyMapping.globVar);"));
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(8.0, co->get());
+}
+
+TEST_F(ControllerScriptEngineLegacyTimerTest, beginTimer_repeatedTimerArrowFunctionCallInClass) {
+    // Single shot timer with minimum allowed interval of 20ms
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(
+            class MyClass {
+               constructor() {
+                  this.timerId = undefined;
+                  this.globVar = 7;
+               }
+               stopTimer() {
+                  engine.stopTimer(this.timerId);
+                  this.timerId = 0;
+                  engine.setValue('[Test]', 'coTimerId', this.timerId + 20);
+               }
+               runTimer() {
+                  this.timerId = engine.beginTimer(20, ()=>this.stopTimer(), false);                  
+                  engine.setValue('[Test]', 'co', this.globVar);      
+                  engine.setValue('[Test]', 'coTimerId', this.timerId);
+               }
+            }
+            var MyMapping = new MyClass();
+            MyMapping.runTimer();)"));
+    processEvents();
+    EXPECT_DOUBLE_EQ(7.0, co->get());
+    double timerId = coTimerId->get();
+    EXPECT_TRUE(timerId > 0);
+
+    cEngine->thread()->msleep(35);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(20, coTimerId->get());
+
+    cEngine->thread()->msleep(35);
+    processEvents();
+
+    EXPECT_DOUBLE_EQ(20, coTimerId->get());
+}
+TEST_F(ControllerScriptEngineLegacyTimerTest, beginTimer_repeatedTimerThisFunctionCallInClass) {
+    // Single shot timer with minimum allowed interval of 20ms
+    EXPECT_TRUE(evaluateAndAssert(
+            R"(
+            class MyClass {
+               constructor() {
+                  this.timerId = undefined;
+                  this.globVar = 7;
+               }
+               stopTimer() {
+                  engine.stopTimer(this.timerId);
+                  this.timerId = 0;
+                  engine.setValue('[Test]', 'coTimerId', this.timerId + 20);
+               }
+               runTimer() {
+                  this.timerId = engine.beginTimer(20, this.stopTimer, false);              
+                  engine.setValue('[Test]', 'co', this.globVar);
+                  engine.setValue('[Test]', 'coTimerId', this.timerId);
+               }
+            }
```

Comment:
This... does not look right. Specifically, the `this` inside the `stopTimer` function will not be an instance of `MyClass`, but will be the [`globalObject()`](https://doc.qt.io/qt-6/qjsvalue.html#call) instead.

This is because `this.stopTimer` is passed as an unbound function on line 859, and `beginTimer` will ultimately use [`QJSValue::call`](https://doc.qt.io/qt-6/qjsvalue.html#call) for invoking the callback.

Reply:
makes sense. it would indeed need to be
```suggestion
            class MyClass {
               constructor() {
                  this.timerId = undefined;
                  this.globVar = 7;
               }
               stopTimer() {
                  engine.stopTimer(this.timerId);
                  this.timerId = 0;
                  engine.setValue('[Test]', 'coTimerId', this.timerId + 20);
               }
               runTimer() {
                  this.timerId = engine.beginTimer(20, this.stopTimer.bind(this), false);              
                  engine.setValue('[Test]', 'co', this.globVar);
                  engine.setValue('[Test]', 'coTimerId', this.timerId);
               }
            }
```